### PR TITLE
EZP-30315 [Travis] Standarize and clean up travis configurations across the repositories

### PR DESCRIPTION
**Pull request by m-tyrala**
Reorganizing .travis.yml file, introducing ezbehat script usage.

Related PRs:
ezsystems/BehatBundle#77
ezsystems/ezplatform-admin-ui#992
ezsystems/ezplatform-page-builder#344
ezsystems/ezplatform-form-builder#162
ezsystems/date-based-publisher#107
ezsystems/ezplatform-admin-ui-modules#189
ezsystems/flex-workflow#128
https://github.com/ezsystems/repository-forms/pull/291

### CODE REVIEW

**Comment by m-tyrala** [comment]
@andrerom on the occasion of this task, I investigated closely the performance differences between Xenial and Trusty distribution on Travis, and it turns out, that in most cases tests takes about 10-20% more time on Xenial dist (you can find the exact time data below). Therefore I removed `dist: xenial` from `.travis.yml` as there is no clear advantage that compensates time losses. Is that change acceptable for you or there are other circumstances which I'm not aware of?

| Trusty time | Xenial time | feature |
|--------------|-------------|-------------|
| 23 min 33 s | 27 min 14 s | ezplatform-admin-ui:ContentTypeFields.feature |
| 36 min 56 s | 42 min 40 s | ezplatform-form-builder:FormBuilderTests.feature |
| 10 min 16 s | 10 min 1 s	| ezplatform-form-builder:FormContentItemAdministration.feature |
| 25 min 19 s | 29 min 8 s | ezplatform-page-builder:Blocks.feature |
| 5 min 3 s | 5 min 23 s | ezplatform-admin-ui:ObjectStates.feature |
| 6 min 22 s | 7 min 10 s | ezplatform-admin-ui:Roles.feature |
| 4 min 9 s | 4 min 56 s | ezplatform-admin-ui:Sections.feature |
| 4 min 19 s | 5 min 14 s | ezplatform-form-builder:FormSubmissionsAdministration.feature |
| 15 min 25 s | 18 min 28 s | ezplatform-form-builder:FormFieldConfiguration.feature |
| 2 min 16 s | 2 min 38 s | ezplatform-admin-ui:Languages.feature |
| 6 min 51 s | 7 min 51 s | ezplatform-admin-ui:ContentDraft.feature |
| 7 min 14 s | 8 min 7 s | ezplatform-admin-ui:ContentManagement.feature |
| 7 min 31 s | 7 min 12 s | ezplatform-page-builder:PageBuilderActions.feature |
| 1 min 47 s | 2 min 12 s | ezplatform-admin-ui:ContentTypeGroup.feature |
| 3 min 45 s | 4 min 28 s | ezplatform-page-builder:ContentIntegration.feature |
| 3 min 18 s | 3 min 46 s | ezplatform-admin-ui:ContentType.feature |
| 3 min 36 s | 4 min 12 s | ezplatform-admin-ui:ContentCreation.feature |
| 1 min 50 s | 2 min 8 s | ezplatform-admin-ui:SystemInfo.feature |
| 9 s | 11 s | ezplatform-admin-ui:Authentication.feature |
| 1 min 2 s | 1 min 10 s | flex-workflow:SendForReviewFromPageBuilder.feature |
| 6 min 59 s | 7 min 49 s | ezplatform-page-builder:ContentScheduler.feature |
| 1 min 8 s | 1 min 14 s | date-based-publisher:PublishInTheFuture.feature |
| 35 s | 42 s | flex-workflow:SendForReview.feature |
| 2 min 59 s | 3 min 6 s | ezplatform-page-builder:Collection.feature |

**Comment by andrerom** [comment]
> Therefore I removed dist: xenial from .travis.yml as there is no clear advantage that compensates time losses. Is that change acceptable for you or there are other circumstances which I'm not aware of?

It's fine, but then rather specify `dist: trusty` to avoid Travis picking what it feels like over time. Aka let's be explicit ;)

**Comment by m-tyrala** [comment]
Sure, not a problem at all.

File: .travis.yml [hunk 1]
```diff
@@ -69,22 +64,33 @@ notifications:
...
-before_script:
+before_install:
+  # Update Docker and Docker Compose
+  - ./bin/.travis/trusty/update_docker.sh
   # Internal auth token dedicated to testing with travis+composer on ezsystems repos, not for reuse!
   - echo "{\"github-oauth\":{\"github.com\":\"d0285ed5c8644f30547572ead2ed897431c1fc09\"}}" > auth.json
+  # Disable XDebug for performance
+  - phpenv config-rm xdebug.ini
+  # Get latest composer build
+  - travis_retry composer selfupdate
```
Comment (andrerom) [inline .travis.yml]: do we need this? Afaik Travis is already doing this now, but maybe that is xenial only.

Comment (m-tyrala) [reply .travis.yml]: According to [documentation](https://docs.travis-ci.com/user/languages/php/#installing-composer-packages) they update composer every 30-60 days and point out that warnings about an outdated program can be ignored. I'm not sure if 30-60 days is enough for us?

Comment (andrerom) [reply .travis.yml]: > I'm not sure if 30-60 days is enough for us?

Afaik last few years it has, we haven't had urgent need for updated version in a long time.

Comment (m-tyrala) [reply .travis.yml]: When composer is up-to-date, then this command takes about 0,5 second, so I'll leave it like this as a low-cost error-proofing.

Comment (andrerom) [reply .travis.yml]: ok

**Comment by m-tyrala** [comment]
@lserwatka ready to merge

**Pull request merged**